### DISCUSSION TITLE
Does Barrier work with Wayland?

### DISCUSSION THREAD

**Discussion by kaihendry** [debauchee/barrier#1781] [Q&A] [answered]
Or is there something specific for Wayland?

**Comment by FilBot3** [comment]
This is also on my radar as well.

From the README

> We will also have our eye on Wayland when the time comes.

Q: Are there any other significant limitations with the current version of Barrier?

    A: Currently:

        Barrier currently has limited UTF-8 support; issues have been reported with processing various languages.
            (see https://github.com/debauchee/barrier/issues/860)
        There is interest in future support for the Wayland compositor/display server protocol ([official site](https://wayland.freedesktop.org/) | [Wikipedia article](https://en.wikipedia.org/wiki/Wayland_(display_server_protocol))) on Linux.
            As of late 2021, there is no expected completion date for Wayland support.
            (see https://github.com/debauchee/barrier/issues/109 and https://github.com/debauchee/barrier/issues/1251 for status or to volunteer your talents)

I don't expect Wayland support for a bit.

**Reply by StuckUnderTheTARDIS** [reply]
Zombie post, rarrrr....

Still can confirm that it doesn't play nicely with Wayland yet on Linux clients.

Cursor & keyboard works, but scrolling is an issue and requires the host to be set to X.Org to get it working again.

Hopefully, this helps anyone that stumbles upon how to get scrolling to work on a Linux client where smooth scrolling isn't an issue.

**Comment by augenix** [comment]
I can confirm that in AlmaLinux 9.3 Wayland (ships as default) session, Barrier server and client can connect and communicate with each other, but mouse and keyboard sharing does not work.  If I switch to a GNOME Classic X.Org(X11) session, then mouse and keyboard share works correctly.  The concern is that X11 is being deprecated and Wayland is the long-term path.  

It also appears that work on Barrier has stopped.  I saw somewhere that a successor is in progress, but I don't remember the name.

**Reply by dev0T** [reply]
If you are aware of a good alternative, please let me know! Barrier is exactly what I was looking for but I am running Plasma 6 with Wayland

**Reply by kaczus** [reply]
https://github.com/input-leap/input-leap/issues/109#issuecomment-1856349330 a fork working with wayland, supposedly it works on gnome, other desktops need to implement support, i believe KDE Plasma 6.1 will work when it's released

**Reply by shibass** [reply]
There's also [lan-mouse](https://github.com/feschber/lan-mouse), which does support wayland but it's quite new and doesn't support X11 or encryption yet

**Comment by mistyron** [comment, marked as answer]
> Or is there something specific for Wayland?

Try input leap, it's a fork and works with Wayland!

**Comment by nbolton** [comment]
# Big Wayland Update! :rocket: 

**Deskflow** v1.16 (free and open source Synergy upstream), Synergy and Input Leap now have Wayland support:
- [**Download Deskflow**](https://github.com/deskflow/deskflow/releases)
- [**Download Input Leap**](https://github.com/input-leap/input-leap/releases/tag/v3.0.0)

Announcement: https://github.com/deskflow/deskflow/discussions/7456

[feschber/lan-mouse](https://github.com/feschber/lan-mouse) (written in Rust) also supports Wayland.

Special thanks to Peter Hutterer (@whot), Olivier Fourdan (@ofourdan), and Povilas Kanapickas (@p12tic) for their work on implementing Wayland support. Special thanks to @GeorgesStavracas for releasing libportal 0.8.0 which introduces input capture and other things needed for Deskflow, Input Leap and Synergy to work on Wayland.
